Documentation/CRDs/ceph-client-crd.md

```diff
@@ -41,3 +41,121 @@ spec:
 ### Prerequisites
 
 This guide assumes you have created a Rook cluster as explained in the main [Quickstart guide](../Getting-Started/quickstart.md).
+
+## Example
+
+### Using a `CephClient` to create Ceph credentials
+
+To use `CephClient` to connect to a Ceph cluster:
+
+#### 1. Create the `CephClient` (`metadata.name` will be your Ceph client ID)
+
+Apply your CephClient resource YAML (see above for an example client config) with `kubectl`:
+
+```console
+kubectl apply -f example.cephclient.yaml
+```
+
+#### 2. Find the generated secret for the `CephClient`
+
+Once your `CephClient` has been processed by Rook (in the rook namespace, for example `rook-ceph`), it will be updated to include your secret:
+
+```console
+kubectl get cephclient exapmle -o jsonpath='{.status.info.secretName}' -n rook-ceph
+```
+
+#### 3. Extract Ceph cluster credentials from the generated secret
+
+Extract Ceph cluster credentials from the generated secret (note that the subkey will be your original client name):
+
+```console
+kubectl get secret rook-ceph-client-example -o jsonpath="{.data.example}" -n rook-ceph | base64 -d
+```
+
+The base64 encoded value that is returned *is* the password for your ceph client.
+
+#### 4. Determine the location of mons for the cluster
+
+To send writes to the cluster, you must retrieve the mons in use:
+
+```console
+kubectl get configmap rook-ceph-mon-endpoints -o jsonpath='{.data.data}' -n rook-ceph | sed 's/.=//g'`
+```
+
+This command *should* produce a line that looks somewhat like this:
+
+```
+XXX.XXX.XXX.XXX:6789,YYY.YYY.YYY.YYY:6789,ZZZ.ZZZ.ZZZ.ZZZ:6789
+```
```

Comment:
This could get out of date over time, but that is not an issue with this documentation. Rather, perhaps @travisn and I should discuss making sure it is possible for apps in other k8s namespaces to reliably connect to the Ceph cluster directly when using the CephClient. There is a fair bit of manual work needed here that could be avoided with some additional feature work. 

I'm not sure if that should be allowing CephClients to be created in arbitrary namespaces, or if there are other solutions.

Reply:
Yeah this is a great point, I look forward to seeing what ya'll decide. 

Originally that's what the issue I posted was about -- trying to get this manual work to zero with a more specific CRD to basically *do* this manual work. That said, at least having it in the documentation I think is great (and was a good learning experience for me), so you should have lots of time!